tostaylo/audio-effects
# 8 add eslint

# What Changed

Wrote an eslint configuration file that was then used in a script. The elsint configuration file is used in a pre-commit hook to find errors. If errors are found, commits will be prohibited until fixed. 

## Design

I installed eslint and configured a json file that defines eslints rules and parameters it uses to scour the code base at the time of pre-commit to check for errors

## Areas Affected

I commented out unused functions in two files and deleted the unused parameter 'audioContext' in the DOM function of dom.js

Commented out functions include
- src/index.js   `// function handlePrefix(prefix) {
//   return (request) => {
//     // compute the default (e.g. / -> index.html)
//     let defaultAssetKey = mapRequestToAsset(request);
//     let url = new URL(defaultAssetKey.url);

//     // strip the prefix from the path for lookup
//     url.pathname = url.pathname.replace(prefix, '/');

//     // inherit all other props from the default request
//     return new Request(url.toString(), defaultAssetKey);
//   };
// }`

- audio.js   `// function getNodeType(node) {
//   switch (node.constructor) {
//     case GainNode: {
//       return 'Gain';
//     }
//   }
// }`

Here is the edit of dom.js (audioContext parameter was removed)
- `export default function DOM({ gainNode, distortionNode }) {
  const volumeControl = document.querySelector('#volume');
  volumeControl.addEventListener(
    'input',
    (e) => handleGain(gainNode, e.target.value),
    false
  );`
## How to Test

test the effectiveness of eslint either by running the script 'npm run eslint' in the command line or by simply committing some code--committing code will prompt a pre-commit hook that initiates a scan by eslint to check for errors.
